src/Microsoft.AspNetCore.Sockets/DefaultConnectionContext.cs

```diff
@@ -36,13 +49,13 @@ public DefaultConnectionContext(string id, Channel<byte[]> transport, Channel<by
 
         public ConnectionStatus Status { get; set; } = ConnectionStatus.Inactive;
 
-        public override string ConnectionId { get; }
+        public override string ConnectionId { get; set; }
```

Comment:
Like most (if not all) the things on the HttpContext, this is settable so that middleware can mutate.